genie-core/src/test/java/com/netflix/genie/core/services/impl/JobCoordinatorServiceImplUnitTests.java

```diff
@@ -1340,4 +1354,46 @@ private JobMetadata getJobMetadata() {
             .withTotalSizeOfAttachments(28080L)
             .build();
     }
+
+
+    /**
+     * Verify the presence of failure tags.
+     * @param argumentCaptor arguments captured in the unit test
+     * @param throwable expected exception
+     */
+    private void verifyFailureTags(final ArgumentCaptor<Map<String, String>> argumentCaptor,
+                                   final Throwable throwable) {
+        final Map<String, String> failureTags = MetricsUtils.newFailureTagsMapForException(throwable);
+
+        Assert.assertTrue(argumentCaptor.getValue().get(MetricsConstants.TagKeys.STATUS).
+            equals(failureTags.get(MetricsConstants.TagKeys.STATUS)));
```

Comment:
If you use `assertEquals(x, y)` rather than `assertTrue(x.equals(y))` the error messages in case of failed assertion will actually show the two things you're comparing and provide a more helpful message.